rewrite-test/src/main/java/org/openrewrite/test/RewriteTest.java

```diff
@@ -423,7 +423,11 @@ default void rewriteRun(Consumer<RecipeSpec> spec, SourceSpec<?>... sourceSpecs)
                             String expected = sourceSpec.noTrim ?
                                     expectedAfter :
                                     trimIndentPreserveCRLF(expectedAfter);
-                            assertThat(actual).as("Unexpected result in \"" + result.getAfter().getSourcePath() + "\"").isEqualTo(expected);
+                            assertThat(actual)
+                                    .as(() -> String.format("Unexpected result in \"%s\"%s",
+                                            result.getAfter().getSourcePath(),
+                                            result.diff().isEmpty() ? "" : "\n" + result.diff()))
```

Comment:
This is not the diff between expected after & actual ; and it seems a tad more difficult the create that using the components going into `Result.diff()` than anticipated. Looking at alternatives now.

Reply:
`org.assertj.core.util.diff.DiffUtils.diff(List, List)` is a bit clunky, but might work; not a fan of the output yet:
```
org.opentest4j.AssertionFailedError: [Unexpected result in "a/b/goodbye.txt"
Changed content at line 1:
expecting:
  ["hello world2"]
but was:
  ["hello world"]
] 
expected: "hello world2"
 but was: "hello world"
Expected :"hello world2"
Actual   :"hello world"
<Click to see difference>
```

Reply:
Can we create a new `Result` object based on the expected and actual afters and use that to create the diff?

Reply:
It's a bit awkward of an API, since it's so specific to diffing SourceFiles in memory using JGit, and the SourceFile could come from any parser. There's also no easy way to do "result.after().withText(actual)` for instance.

Reply:
I think at most you could do something like
```java
assertThat(actual)
        .as(() -> {
            SourceFile after = result.getAfter();
            PlainText actualSourceFile = PlainText.builder()
                    .id(after.getId())
                    .charsetBomMarked(after.isCharsetBomMarked())
                    .fileAttributes(after.getFileAttributes())
                    .charsetName(after.getCharset().name())
                    .checksum(after.getChecksum())
                    .markers(after.getMarkers())
                    .sourcePath(after.getSourcePath())
                    .text(actual)
                    .build();
            String diff = new Result(after, actualSourceFile, emptyList()).diff();
            return String.format("Unexpected result in \"%s\"%s",
                    after.getSourcePath(),
                    diff.isEmpty() ? "" : "\n" + diff);
        })
        .isEqualTo(expected);
```
but that's not really what I'd hope to do here.